operations/mimir/servicemonitor.libsonnet

```diff
@@ -0,0 +1,51 @@
+{
+  local this = self,
+
+  serviceMonitor: if $._config.create_service_monitor then {
+    apiVersion: 'monitoring.coreos.com/v1',
+    kind: 'ServiceMonitor',
+    metadata: {
+      name: 'mimir',
+      namespace: $._config.namespace,
+    },
+    spec: {
+      namespaceSelector: {
+        matchNames: [$._config.namespace],
+      },
+      servicesToRelabel:: [
+        'compactor',
+        'distributor',
+        'ingester',
+        'querier',
+        'query-front',
+        'store-gateway',
+      ],
+      endpoints: [
+        {
+          port: port.name,
+          path: '/metrics',
+          [if std.member(self.servicesToRelabel, this[name].spec.selector.name) then 'relabelings']: [{
+            sourceLabels: ['job'],
+            action: 'replace',
+            replacement: $._config.namespace + '/$1',
+            targetLabel: 'job',
+          }],
```

Comment:
i think i'm missing that this relabelling rule does. What is the original job name in prometheus?

Reply:
the job is ingester/querier/... , the rule prepends the namespace.

I don't usually do this, but it's already present in the [chart](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/templates/lib/service-monitor.tpl#L49-L52) and in loki jsonnet, so it kept it